rules/repositories.bzl

```diff
@@ -186,21 +186,17 @@ def _rules_ios_bzlmod_dependencies():
         # you'd like to set it. If you want to use a different rules_apple than
         # we have here - define rules_apple_api - to the value you'd like
         _maybe(
-            github_repo,
```

Comment:
@luispadron the `github_repo` is a helper function to avoid having to form the URL that makes an http request - is there a reason to change this - it's used throughout the file

Reply:
Mainly it's [used throughout](https://github.com/bazel-ios/rules_ios/pull/784/files#diff-435e1309098a3a6ba8d1758ed935645d702c31d66bbf05b6f7d026a3d82fdfdcL154) the file to load deps so it is odd to have it not using it in this one place

Reply:
yes there is, that would download an on-demand archive and not a release asset. So not only is it going to download things it doesnt need but its susceptible to the github archive sha issue.

Reply:
in general we should prefer release assets over on-demand archives: https://blog.bazel.build/2023/02/15/github-archive-checksum.html

Reply:
The `github_repo` rule predates the incident from github - I can't find their official comms from github but only this: https://github.blog/changelog/2023-01-30-git-archive-checksums-may-change/ which likely breaks most of the internet outside of bazel

Reply:
Sure, I guess I'm confused what the issue is here? We were using the release assets before you merged the rules_apple 3 change and I left feedback in your PR about it which went unaddressed. 

github_repo will request the on-demand archive of the entire repository, including fixtures and other things which are completely unnecessary to pull down here. 

These tagged release assets are provided by the rule authors to contain the stuff needed to use the rules, so on top of it being lighter to download it also won't ever be effected by that git sha issue and it's what bzlmod would download anyway.

Reply:
@jerrymarino the entries before #773 looked like:

```
         _maybe(
             http_archive,
             name = "build_bazel_rules_swift",
             sha256 = "3a595a64afdcaf65b74b794661556318041466d727e175fa8ce20bdf1bb84ba0",
             url = "https://github.com/bazelbuild/rules_swift/releases/download/1.10.0/rules_swift.1.10.0.tar.gz",
         )

         _maybe(
             http_archive,
             name = "build_bazel_rules_apple",
             sha256 = "8ac4c7997d863f3c4347ba996e831b5ec8f7af885ee8d4fe36f1c3c8f0092b2c",
             url = "https://github.com/bazelbuild/rules_apple/releases/download/2.5.0/rules_apple.2.5.0.tar.gz",
         )
```